### PR TITLE
minim asset count

### PR DESCRIPTION
**Pull request by N1ghtStorm**

### CODE REVIEW

File: pallets/substrate-app/src/lib.rs [hunk 1]
```diff
@@ -472,6 +485,40 @@ pub mod pallet {
             );
             Ok(())
         }
+
+        #[pallet::call_index(9)]
+        #[pallet::weight(<T as Config>::WeightInfo::register_erc20_asset())]
```
Comment (wer1st) [inline pallets/substrate-app/src/lib.rs:505]: You use `register_erc20_asset()` for weight everywhere, is it ok?
If you plan to replace it with associated weights, please add `TODO` comment

File: pallets/substrate-app/src/lib.rs [hunk 2]
```diff
@@ -472,6 +485,40 @@ pub mod pallet {
...
+        pub fn set_minimum_xcm_incoming_asset_count(
```
Comment (wer1st) [inline pallets/substrate-app/src/lib.rs:506]: And what about unit tests / migrations / benchmarks ?

File: pallets/types/src/substrate.rs [hunk 3]
```diff
@@ -97,7 +97,12 @@ pub enum XCMAppCall {
         asset_id: MainnetAssetId,
         sidechain_asset: ParachainAssetId,
         asset_kind: AssetKind,
+        min_amount: u128,
```
Comment (wer1st) [inline pallets/types/src/substrate.rs]: Do you have special type for balance like `Balance` here? If yes, please replace it with `u128`

File: pallets/substrate-app/src/lib.rs (hunk 1 above)
Comment (N1ghtStorm) [reply pallets/substrate-app/src/lib.rs:505]: it will be done in a separate task

File: pallets/types/src/substrate.rs (hunk 3 above)
Comment (N1ghtStorm) [reply pallets/types/src/substrate.rs]: the XCMApp pallet has no infrmation about Sora2 balances, the conversion is performed in substrateApp

File: pallets/substrate-app/src/lib.rs (hunk 2 above)
Comment (N1ghtStorm) [reply pallets/substrate-app/src/lib.rs:506]: it will be done in a separate task

File: pallets/substrate-app/src/lib.rs [hunk 4]
```diff
@@ -350,6 +350,7 @@ pub mod pallet {
             asset_id: AssetIdOf<T>,
             sidechain_asset: ParachainAssetId,
             allowed_parachains: Vec<u32>,
+            min_amount: BalanceOf<T>,
```
Comment (Alexey-N-Chernyshov) [inline pallets/substrate-app/src/lib.rs]: It's not clear from the name that it is a minimal xcm transferable amount. I suggest to rename to `minimal_xcm_amount`.

File: pallets/substrate-app/src/lib.rs [hunk 5]
```diff
@@ -482,6 +529,7 @@ pub mod pallet {
             asset_kind: AssetKind,
             sidechain_precision: u8,
             allowed_parachains: Vec<u32>,
+            min_amount: u128,
```
Comment (Alexey-N-Chernyshov) [inline pallets/substrate-app/src/lib.rs]: SAP naming not clear.

**Review by Alexey-N-Chernyshov — CHANGES_REQUESTED** [review]

File: pallets/substrate-app/src/lib.rs (hunk 4 above)
Comment (N1ghtStorm) [reply pallets/substrate-app/src/lib.rs]: Done

File: pallets/substrate-app/src/lib.rs (hunk 5 above)
Comment (N1ghtStorm) [reply pallets/substrate-app/src/lib.rs]: Done

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: pallets/types/src/substrate.rs (hunk 3 above)
Comment (vovac12) [reply pallets/types/src/substrate.rs]: @N1ghtStorm you can use `MainnetBalance` type 

And I think it's better to rename this field to `minimal_xcm_amount`

File: pallets/substrate-app/src/lib.rs (hunk 1 above)
Comment (vovac12) [reply pallets/substrate-app/src/lib.rs:505]: I think it's good to have `TODO` here as @wer1st suggested

**Pull request merged**